worlds/mmx3/Regions.py

```diff
@@ -0,0 +1,514 @@
+import typing
+
+from BaseClasses import CollectionState, MultiWorld, Region, Entrance, ItemClassification
+from .Locations import MMX3Location
+from .Items import MMX3Item
+from .Names import LocationName, ItemName, RegionName, EventName
+from worlds.AutoWorld import World
+
+
+def create_regions(multiworld: MultiWorld, player: int, world: World, active_locations):
```

Comment:
```suggestion
if typing.TYPE_CHECKING:
    from . import MMX3World


def create_regions(world: "MMX3World", active_locations):
    multiworld = world.multiworld
    player = world.player
```

If you do that weird type_checking thing, you can import your actual world class for your typing. It does need to be put in quotes when typing.
And then, you can get the multiworld and player directly from the world class.